features/mbedtls/README.md

```diff
@@ -1,14 +1,96 @@
+
+
+Configuring Mbed TLS features
+-----------------------------
+
+Mbed TLS makes it easy to disable any feature during compilation, if that feature isn't required for a particular project. The default configuration enables all modern and widely-used features of the TLS protocol, which should meet the needs of most projects. It also disables all older and less common features, to minimize the code footprint.
+
+The list of available compilation flags is available in the fully documented [config.h file](https://github.com/ARMmbed/mbedtls/blob/development/include/mbedtls/config.h).
+
+If you need to adjust those flags, you can provide your own supplementary configuration-adjustment file with suitable `#define` and `#undef` statements. These will be included between the default definitions and the sanity checks. Your configuration file should be in your application's include directory, and can be named freely; you just need to let mbed TLS know the file's name. To do that, you can use the [Mbed OS Configuration
+system](https://docs.mbed.com/docs/mbed-os-api/en/latest/api/md_docs_config_system.html)
+
+For example, if you wanted to enable the options, `MBEDTLS_PEM_WRITE_C` and `MBEDTLS_CMAC_C`, and provide your own additional configuration file for Mbed TLS named `my_config.h`, you could define these in a top level `mbed_app.json` configuration file in the root directory of your project.
+
+The Mbed TLS configuration file would be specified in the `.json` file as follows.
+
+```
+{
+    "macros" : [
+
+        "MBEDTLS_USER_CONFIG_FILE" : "my_config.h",
+
+        "MBEDTLS_PEM_WRITE_C",
+        "MBEDTLS_CMAC_C"
+    ]
+    ....remainder of file...
+}
+```
+
+The additional configuration file, `my_config.h`, can then be used as a normal configuration header file to include additional configurations. For example, it could include the follow lines to include ECJPAKE, and the CBC block mode:
+
+```
+    #define MBEDTLS_ECJPAKE_C
+    #define MBEDTLS_KEY_EXCHANGE_ECJPAKE_ENABLED
+
+    #undef MBEDTLS_CIPHER_MODE_CBC
+```
+
+## Getting Mbed TLS from GitHub
+
+Mbed TLS is maintained and developed in the open, independently of Mbed OS, and its source can be found on GitHub here: [ARMmbed/mbedtls](https://github.com/ARMmbed/mbedtls). As a consequence, unlike other parts of mbed OS, changes to Mbed TLS must be committed upstream in the Mbed TLS repository.
+
+To import into an instance of Mbed OS a different version of Mbed TLS, a `Makefile` script is provided to update the local git repository, extract a specific version, and to modify the configuration files to those used for the Mbed OS defaults.
+
+To use the `Makefile`, you can either set `MBED_TLS_RELEASE` environment variable to the git tag or commit id of the Mbed TLS Release or version you want to use, or alternatively you can modify the `Makefile` itself.
```

Comment:
Tag or commit? Not a branch name to get the tip of the branch? Can't I leave it blank to get the default, and would I get the latest release [master], the latest beta [development], or some untested experimental thing [we don't have this]?